source/module_hamilt_general/module_xc/xc_functional_libxc.h

```diff
@@ -0,0 +1,99 @@
+#ifndef XC_FUNCTIONAL_LIBXC_H
+#define XC_FUNCTIONAL_LIBXC_H
+
+#ifdef USE_LIBXC
+
+#include "module_base/matrix.h"
+#include "module_base/vector3.h"
+
+#include <xc.h>
+
+#include <tuple>
+#include <vector>
+
+	class Charge;

```

Comment:
do we need this class?

Reply:
I think this is just a declaration to avoid the "error: unknown type name 'Charge'" error when compiling, as the following functions in this file need this as a parameter type.

Reply:
For the function parameter which is just a pointer `Charge*` or a reference `Charge&`, it's better to write `class Charge` before the function instead of `#include "charge.h"` in the header files.